src/main/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionListener.java

```diff
@@ -12,28 +14,53 @@
 public abstract class FlowExecutionListener implements ExtensionPoint {
 
     /**
-     * Called when a {@link FlowExecution} has started running or resumed.
+     * Called when a {@link FlowExecution} has started running.
      *
-     * @param execution The {@link FlowExecution} that has started running or resumed.
-     * @param resumed True the execution is resuming, false if it's starting for the first time.
+     * The {@link FlowExecution} will already have been added to the {@link FlowExecutionList} by this point.
+     *
+     * @param execution The {@link FlowExecution} that has started running.
+     */
+    public void onRunning(FlowExecution execution) {
+    }
+
+    /**
+     * Called when a {@link FlowExecution} has resumed.
+     *
+     * The {@link FlowExecution} will already have been added to the {@link FlowExecutionList} by this point.
+     *
+     * @param execution The {@link FlowExecution} that has resumed.
      */
-    public void onRunning(FlowExecution execution, boolean resumed) {
+    public void onResumed(FlowExecution execution) {
     }
 
     /**
      * Called when a {@link FlowExecution} has completed.
      *
+     * The {@link FlowExecution} will already have been removed from the {@link FlowExecutionList} by this point,
+     * {@link GraphListener.Synchronous#onNewHead(FlowNode)} will have already been called for the {@link FlowEndNode},
+     * {@link FlowEndNode#getResult()} will return non-null, and if the Pipeline has failed,
```

Comment:
Possibly misleading, since `FlowEndNode.result` is `final`, so it is not like it would have returned null at some earlier point—it is just that the `FlowEndNode` is now available at all, and BTW you might be interested in this method.

Would be good to add `@Nonnull` annotations to `FlowEndNode`.